qiita_db/study.py

```diff
@@ -142,18 +142,42 @@ class Study(QiitaStatusObject):
     """
     _table = "study"
     # The following columns are considered not part of the study info
-    _non_info = {"email", "study_status_id", "study_title"}
+    _non_info = {"email", "study_title"}
 
     def _lock_non_sandbox(self, conn_handler):
         """Raises QiitaDBStatusError if study is non-sandboxed"""
         if self.status != 'sandbox':
             raise QiitaDBStatusError("Illegal operation on non-sandbox study!")
 
-    def _status_setter_checks(self, conn_handler):
-        r"""Perform a check to make sure not setting status away from public
-        """
-        if self.check_status(("public", )):
-            raise QiitaDBStatusError("Illegal operation on public study!")
+    @property
+    def status(self):
+        r"""The status is inferred by the status of its processed data"""
+        conn_handler = SQLConnectionHandler()
+        # Get the status of all its processed data
+        sql = """SELECT processed_data_status
+                FROM qiita.processed_data_status pds
+                  JOIN qiita.processed_data pd
+                    ON pds.processed_data_status_id=pd.processed_data_status_id
+                  JOIN qiita.study_processed_data spd
+                    ON spd.processed_data_id=pd.processed_data_id
+                WHERE spd.study_id = %s"""
+        pd_statuses = conn_handler.execute_fetchall(sql, (self._id,))
+
+        if not pd_statuses:
+            # If there are no processed data, then the status is sandbox
+            study_status = 'sandbox'
+        else:
+            pd_statuses = set(s[0] for s in pd_statuses)
+            if 'public' in pd_statuses:
```

Comment:
If we are hiding the non-public procesed data this makes sense to have it in this order here.